src/libraries/System.Security.Cryptography/tests/X509Certificates/X500DistinguishedNameBuilderTests.cs

```diff
@@ -0,0 +1,379 @@
+            AssertExtensions.Throws<ArgumentException>(
+                "encodedValue",
+                () => builder.AddEncoded(new Oid(TestOid), new byte[] { 0xFF, 0xFF, 0xFF }));
+            AssertExtensions.Throws<ArgumentException>(
+                "encodedValue",
+                () => builder.AddEncoded(new Oid(TestOid), stackalloc byte[] { 0xFF, 0xFF, 0xFF }));
+        }
+
+        [Fact]
+        public static void AddEncoded_StringOid_NullOrEmpty_Fails()
+        {
+            X500DistinguishedNameBuilder builder = new();
+            AssertExtensions.Throws<ArgumentNullException>(
+                "oidValue",
+                () => builder.AddEncoded((string)null, Array.Empty<byte>()));
+            AssertExtensions.Throws<ArgumentException>(
+                "oidValue",
+                () => builder.AddEncoded("", Array.Empty<byte>()));
+        }
+
+        [Fact]
+        public static void AddEncoded_Oid_NullOrEmptyValue_Fails()
+        {
+            X500DistinguishedNameBuilder builder = new();
+            AssertExtensions.Throws<ArgumentNullException>(
+                "oid",
+                () => builder.AddEncoded((Oid)null, Array.Empty<byte>()));
+            AssertExtensions.Throws<ArgumentException>(
+                "oid",
+                () => builder.AddEncoded(new Oid(), Array.Empty<byte>()));
+            AssertExtensions.Throws<ArgumentException>(
+                "oid",
+                () => builder.AddEncoded(new Oid("", null), Array.Empty<byte>()));
+        }
+
+        [Fact]
+        public static void AddEncoded_Value_Null_Fails()
+        {
+            X500DistinguishedNameBuilder builder = new();
+            AssertExtensions.Throws<ArgumentNullException>(
+                "encodedValue",
+                () => builder.AddEncoded(TestOid, (byte[])null));
+            AssertExtensions.Throws<ArgumentNullException>(
+                "encodedValue",
+                () => builder.AddEncoded(new Oid(TestOid, null), (byte[])null));
+        }
+
+        [Fact]
+        public static void Add_InvalidUniversalTagNumber()
+        {
+            X500DistinguishedNameBuilder builder = new();
+            AssertExtensions.Throws<ArgumentException>(
+                "stringEncodingType",
+                () => builder.Add(TestOid, "True", UniversalTagNumber.Boolean));
+            AssertExtensions.Throws<ArgumentException>(
+                "stringEncodingType",
+                () => builder.Add(new Oid(TestOid, null), "True", UniversalTagNumber.Boolean));
+        }
+
+        [Fact]
+        public static void Add_OidString_NullOrEmpty_Fails()
+        {
+            X500DistinguishedNameBuilder builder = new();
+            AssertExtensions.Throws<ArgumentNullException>(
+                "oidValue",
+                () => builder.Add((string)null, "banana"));
+            AssertExtensions.Throws<ArgumentException>(
+                "oidValue",
+                () => builder.Add("", "banana"));
```

Comment:
I used to love "banana" just because of the "anana".  Now I've seen Horrible Histories and love it even more.

https://youtu.be/zPtYmq5qFVA?t=5.  You're welcome :)